autobahn/main.go

```diff
@@ -0,0 +1,268 @@
+			srvFailed  bool
+			hdrWritten bool
+			counter    statusCounter
+		)
+
+		var cases []string
+		for id := range report[server] {
+			cases = append(cases, id)
+		}
+		sortBySegment(cases)
+		for _, id := range cases {
+			c := report[server][id]
+
+			var r entryReport
+			err := decodeFile(path.Join(base, c.ReportFile), &r)
+			if err != nil {
+				log.Fatal(err)
+			}
+			counter.Inc(c.Behavior)
+			bad := failing(c.Behavior)
+			if bad {
+				srvFailed = true
+				failed = true
+			}
+			if *verbose || bad {
+				if !hdrWritten {
+					hdrWritten = true
+					n, _ := fmt.Fprintf(os.Stderr, "AGENT %q\n", server)
+					fmt.Fprintf(tw, "%s\n", strings.Repeat("=", n-1))
+				}
+				fmt.Fprintf(tw, "%s\t%s\t%s\n", server, id, c.Behavior)
+			}
+			if bad {
+				fmt.Fprintf(tw, "\tdesc:\t%s\n", r.Description)
+				fmt.Fprintf(tw, "\texp: \t%s\n", r.Expectation)
+				fmt.Fprintf(tw, "\tact: \t%s\n", r.Result)
+			}
+		}
+		if hdrWritten {
+			fmt.Fprint(tw, "\n")
+		}
+		var status string
+		if srvFailed {
+			status = statusFailed
+		} else {
+			status = statusOK
+		}
+		n, _ := fmt.Fprintf(tw, "AGENT %q SUMMARY (%s)\n", server, status)
+		fmt.Fprintf(tw, "%s\n", strings.Repeat("=", n-1))
+
+		fmt.Fprintf(tw, "TOTAL:\t%d\n", counter.Total)
+		fmt.Fprintf(tw, "%s:\t%d\n", statusOK, counter.OK)
+		fmt.Fprintf(tw, "%s:\t%d\n", statusInformational, counter.Informational)
+		fmt.Fprintf(tw, "%s:\t%d\n", statusUnimplemented, counter.Unimplemented)
+		fmt.Fprintf(tw, "%s:\t%d\n", statusNonStrict, counter.NonStrict)
+		fmt.Fprintf(tw, "%s:\t%d\n", statusUnclean, counter.Unclean)
+		fmt.Fprintf(tw, "%s:\t%d\n", statusFailed, counter.Failed)
+		fmt.Fprint(tw, "\n")
+		tw.Flush()
+	}
+	var rc int
+	if failed {
+		rc = 1
+		fmt.Fprintf(tw, "\n\nTEST %s\n\n", statusFailed)
+	} else {
+		fmt.Fprintf(tw, "\n\nTEST %s\n\n", statusOK)
+	}
+
+	tw.Flush()
+	os.Exit(rc)
+}
+
+type spec struct {
+	OutDir string `json:"outdir"`
+}
+
+type report map[string]server
+
+type server map[string]entry
+
+type entry struct {
+	Behavior        string `json:"behavior"`
+	BehaviorClose   string `json:"behaviorClose"`
+	Duration        int    `json:"duration"`
+	RemoveCloseCode int    `json:"removeCloseCode"`
+	ReportFile      string `json:"reportFile"`
+}
+
+type entryReport struct {
+	Description string `json:"description"`
+	Expectation string `json:"expectation"`
+	Result      string `json:"result"`
+	Duration    int    `json:"duration"`
+}
+
+func decodeFile(path string, x interface{}) error {
+	f, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	d := json.NewDecoder(f)
+	return d.Decode(x)
+}
+
+func compareBySegment(a, b string) int {
+	as := strings.Split(a, ".")
+	bs := strings.Split(b, ".")
+	for i := 0; i < min(len(as), len(bs)); i++ {
+		ax := mustInt(as[i])
+		bx := mustInt(bs[i])
+		if ax == bx {
+			continue
+		}
+		return ax - bx
+	}
+	return len(b) - len(a)
+}
+
+func mustInt(s string) int {
+	const bits = 32 << (^uint(0) >> 63)
```

Comment:
hm...32 ?

Reply:
the same question

Reply:
What is the question? =)
32 or 64 bits for integer parsing.